src/coreclr/vm/peassembly.cpp

```diff
@@ -869,46 +860,14 @@ PEAssembly *PEAssembly::Create(PEAssembly *pParentAssembly,
     // we have.)
     SafeComHolder<IMetaDataEmit> pEmit;
     pAssemblyEmit->QueryInterface(IID_IMetaDataEmit, (void **)&pEmit);
-    RETURN new PEAssembly(NULL, pEmit, pParentAssembly, FALSE);
+    RETURN new PEAssembly(NULL, pEmit, FALSE);
 }
 
 #endif // #ifndef DACCESS_COMPILE
 
 
 #ifndef DACCESS_COMPILE
 
-// ------------------------------------------------------------
-// Descriptive strings
-// ------------------------------------------------------------
-
-// Effective path is the path of nearest parent (creator) assembly which has a nonempty path.
-
-const SString &PEAssembly::GetEffectivePath()
-{
-    CONTRACTL
-    {
-        INSTANCE_CHECK;
-        NOTHROW;
-        GC_NOTRIGGER;
-        MODE_ANY;
-    }
-    CONTRACTL_END;
-
-    PEAssembly* pPEAssembly = this;
-
-    while (pPEAssembly->m_PEImage == NULL
-        || pPEAssembly->m_PEImage->GetPath().IsEmpty())
-    {
-        if (pPEAssembly->m_creator)
-            pPEAssembly = pPEAssembly->m_creator;
-        else // Unmanaged exe which loads byte[]/IStream assemblies
-            return SString::Empty();
-    }
-
-    return pPEAssembly->m_PEImage->GetPath();
-}
-
-
 // Codebase is the fusion codebase or path for the assembly.  It is in URL format.
 // Note this may be obtained from the parent PEAssembly if we don't have a path or fusion
 // assembly.
```

Comment:
Are we slightly changing the behavior here? We will now only return codebase for assemblies loaded from individual files. Maybe need to update the comment.

Reply:
I have updated the comment here and also fixed the error message for dynamic assemblies. The exception type was right so the tests were passing, but the error message was bogus.